modules/optional/option.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package optional
+
+type Option[T any] []T
+
+func None[T any]() Option[T] {
+	return nil
+}
+
+func Some[T any](v T) Option[T] {
```

Comment:
Which language?
I've never seen that notation.

Reply:
https://en.wikipedia.org/wiki/Option_type
https://dotnet.github.io/dotNext/features/core/optional.html
https://developer.apple.com/documentation/swift/optional
https://github.com/nlkl/Optional

Reply:
More thoughts from my opinion:

I can understand that many languages use "Some", but most languages using "Some" because they have "match" pattern like :

```
value = option.Match(
  some: ...
```

For languages like C++, which don't have "match" pattern , they never use "some", they only use "value": https://en.cppreference.com/w/cpp/utility/optional/optional

And Go is more C-style language, no match pattern.

To me, either is fine. I am pretty neutral.